File: klasa 2/python/euklides.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-
#
#  euklides.py
#

def euklides(a,b):
    
    while a != b:
        if a > b:
            a = a -b
        else:
            b = b - a
            
    return a


def nww(a,b):
    
    pass


def main(args):
    
    a = int(input("Podaj pierwszą liczbę: "))
    b = int(input("Podaj drugą liczbę: "))
    
    print("NWD({}, {}) = {}".format(a, b, euklides(a,b)))
    
    return 0

if __name__ == '__main__':
    import sys
    sys.exit(main(sys.argv))

File: klasa 2/c++/euklides.cpp
/*
 * euklides.cpp

 */


#include <iostream>

using namespace std;

int nwd_optymalny(int a, int b)
{   
    int licznik = 0;
    
    while (a > 0) 
    {
        a = a % b;
        b = b - a;
        licznik++;
    }
    
    cout << "Liczba powtórzeń: " << licznik << endl;
    
    return b;
}

int nwd_klasyczny(int a, int b)
{
    int licznik = 0;
    
    while(a!=b)
	{
		if (a > b)
        {
			a = a - b;
            licznik++;
		}
        else
        {
			b = b - a;
            licznik++;
        }
    }
    
    cout << "Liczba powtórzeń: " << licznik << endl;
    
    return a;
}

int main(int argc, char **argv)
{
	int a, b;
    a = b = 0;
    
    cout << "Podaj liczbę: ";
    cin >> a;
            
    cout << "Podaj drugą liczbę: ";
    cin >> b;

    cout << nwd_klasyczny(a, b) << endl;
    cout << nwd_optymalny(a, b);

	return 0;
}

